images/scripts/rhel.setup

```diff
@@ -291,6 +291,19 @@ fi
 pkgs="$TEST_PACKAGES $COCKPIT_DEPS $IPA_CLIENT_PACKAGES"
 $YUM_INSTALL $pkgs
 
+# If the insights-client is installed, update the insights-client
+# "egg" and make sure that only the newest one is used.
+
+if [ -x /usr/bin/insights-client ]; then
+    insights-client --status >/dev/null
+    if [ -f /var/lib/insights-client/newest.egg ]; then
+        cp /var/lib/insights-client/newest.egg /var/lib/insights-client/last_stable.egg
+        cp /var/lib/insights-client/newest.egg.asc /var/lib/insights-client/last_stable.egg.asc
+        rm -f /etc/insights-client/rpm.egg /etc/insights-client/rpm.egg.asc
```

Comment:
That could do with another comment -- so apparently there's a "newest.egg" which is being backed up, and an "rpm.egg" which is removed -- what are these two? Why is newest.egg precious and rpm.egg isn't?

Reply:
Normally, `newest.egg` is copied automatically to `last_stable.egg` once it has successfully been used once.

But, the idea is that we _only_ want to ever run newest.egg and if it is broken, we want to know about that, instead of having the insights-client fall back to an older egg.  That's the real reason to promote newest.egg to last_stable.egg unconditionally.  Removing rpm.egg takes that fallback out of the equation, too.

Also, `insights-client --version` only looks at last_stable.egg or rpm.egg, never at newest.egg.  So copying newest.egg to last_stable.egg makes that work as expected, too.

(So, yeah. this could do with another comment. :-)